### PR TITLE
Cli personal access token

### PR DESCRIPTION
**Pull request by akhil-jha**
```
(env) [akjha@akjha robottelo]$ pytest tests/foreman/cli/test_user.py::TestPersonalAccessToken
==================================================================== test session starts =====================================================================
platform linux -- Python 3.8.10, pytest-6.2.4, py-1.10.0, pluggy-0.13.1
shared_function enabled - OFF - scope:  - storage: file
rootdir: /home/akjha/satelliteqe/robottelo, configfile: pyproject.toml
plugins: reportportal-5.0.8, forked-1.3.0, mock-3.6.1, xdist-2.3.0, cov-2.12.0, services-2.2.1, ibutsu-1.16
collected 3 items                                                                                                                                            

tests/foreman/cli/test_user.py ...                                                                                                                     [100%]

=============================================================== 3 passed in 204.49s (0:03:04) ================================================================

```

### CODE REVIEW

**Comment by akhil-jha** [comment]
trigger: test-robottelo
pytest: -k tests/foreman/cli/test_user.py::TestPersonalAccessToken

**Comment by akhil-jha** [comment]
trigger: test-robottelo
pytest: tests/foreman/cli/test_user.py::TestPersonalAccessToken

File: robottelo/cli/user.py [hunk 1]
```diff
@@ -93,3 +94,21 @@ def ssh_keys_info(cls, options=None):
         """
         cls.command_sub = 'ssh-keys info'
         return cls.execute(cls._construct_command(options), output_format='csv')
+
+    @classmethod
+    def personal_access_token_create(cls, options=None):
```
Comment (JacobCallahan) [inline robottelo/cli/user.py:99]: Could you just call this `access_token_create`?

File: robottelo/cli/user.py [hunk 2]
```diff
@@ -93,3 +94,21 @@ def ssh_keys_info(cls, options=None):
...
+        """
+        Usage:
+        hammer user access-token create [ARG] ...
+        """
+        cls.command_sub = 'access-token create'
+        return cls.execute(cls._construct_command(options), output_format='csv')
+
+    @classmethod
+    def personal_access_token_revoke(cls, options=None):
```
Comment (JacobCallahan) [inline robottelo/cli/user.py:108]: Similar question

File: tests/foreman/cli/test_user.py [hunk 3]
```diff
@@ -405,9 +407,21 @@ def test_personal_access_token_admin(self):
 
         :CaseImportance: High
         """
+        user = make_user({'admin': '1'})
+        token_name = gen_alphanumeric()
+        result = User.personal_access_token_create({'name': token_name, 'user-id': user['id']})
+        token_value = result[0]['message'].split('\n')[-1]
+        curl_command = (
+            f'curl -k -u {user["login"]}:{token_value} '
+            f'https://{settings.server.hostname}/api/v2/users'
```
Comment (JacobCallahan) [inline tests/foreman/cli/test_user.py:416]: If you inherit the default_sat fixture, you can use it to not only execute commands, but you can reference it's hostname directly and no longer need to import `command`

**Review by JacobCallahan — CHANGES_REQUESTED** [review]

**Comment by akhil-jha** [comment]
**Results after changes:**
```
(env) [akjha@akjha robottelo]$ pytest tests/foreman/cli/test_user.py::TestPersonalAccessToken
==================================================================== test session starts =====================================================================
platform linux -- Python 3.8.10, pytest-6.2.4, py-1.10.0, pluggy-0.13.1
shared_function enabled - OFF - scope:  - storage: file
rootdir: /home/akjha/satelliteqe/robottelo, configfile: pyproject.toml
plugins: reportportal-5.0.8, forked-1.3.0, mock-3.6.1, xdist-2.3.0, cov-2.12.0, services-2.2.1, ibutsu-1.16
collected 3 items                                                                                                                                            

tests/foreman/cli/test_user.py ...                                                                                                                     [100%]

=============================================================== 3 passed in 131.36s (0:02:11) ================================================================

```

**Comment by akhil-jha** [comment]
trigger: test-robottelo
pytest: tests/foreman/cli/test_user.py::TestPersonalAccessToken

File: tests/foreman/cli/test_user.py [hunk 4]
```diff
@@ -448,3 +480,26 @@ def test_expired_personal_access_token(self):
 
         :CaseImportance: Medium
         """
+        user = make_user()
+        User.add_role({'login': user['login'], 'role': 'View hosts'})
+        token_name = gen_alphanumeric()
+        datetime_now = datetime.datetime.utcnow()
+        datetime_expire = datetime_now + datetime.timedelta(seconds=20)
```
Comment (devendra104) [inline tests/foreman/cli/test_user.py:487]: In Test step, it is mentioned 1 minute?

File: tests/foreman/cli/test_user.py [hunk 5]
```diff
@@ -429,10 +445,26 @@ def test_positive_personal_access_token_user_with_role(self):
 
         :CaseImportance: High
         """
+        user = make_user()
```
Comment (devendra104) [inline tests/foreman/cli/test_user.py:448]: (Optional): Can we create a common fixture for user, role, and token?

File: tests/foreman/cli/test_user.py (hunk 4 above)
Comment (akhil-jha) [reply tests/foreman/cli/test_user.py:487]: Ah my bad. When I wrote the stubbed cases I thought minimum is 1 minute. But while writing it I came across seconds.

File: tests/foreman/cli/test_user.py (hunk 5 above)
Comment (akhil-jha) [reply tests/foreman/cli/test_user.py:448]: Keeping it as an improvement. Will be adding new scenarios as well

File: robottelo/cli/user.py (hunk 1 above)
Comment (akhil-jha) [reply robottelo/cli/user.py]: Done

File: robottelo/cli/user.py (hunk 2 above)
Comment (akhil-jha) [reply robottelo/cli/user.py]: Done

File: tests/foreman/cli/test_user.py (hunk 3 above)
Comment (akhil-jha) [reply tests/foreman/cli/test_user.py]: Done

**Review by devendra104 — APPROVED** [review]
LGTM

**Comment by devendra104** [comment]
@akhil-jha Just a suggestion, no need to put a new commit for every change. I think a single commit message is sufficient for a PR until some major changes do not happen.

File: robottelo/cli/user.py [hunk 6]
```diff
@@ -93,3 +94,14 @@ def ssh_keys_info(cls, options=None):
         """
         cls.command_sub = 'ssh-keys info'
         return cls.execute(cls._construct_command(options), output_format='csv')
+
+    @classmethod
+    def personal_access_token(cls, action=None, options=None):
```
Comment (JacobCallahan) [inline robottelo/cli/user.py:99]:
```suggestion
    def access_token(cls, action=None, options=None):
```
This was my original suggestion

**Review by JacobCallahan — CHANGES_REQUESTED** [review]

**Comment by akhil-jha** [comment]
trigger: test-robottelo
pytest: tests/foreman/cli/test_user.py::TestPersonalAccessToken

File: robottelo/cli/user.py (hunk 6 above)
Comment (akhil-jha) [reply robottelo/cli/user.py]: My bad. 
Changes done

**Comment by devendra104** [comment]
@JacobCallahan Can you give a slight look at this MR, It seems @akhil-jha addressed all of your feedback?

**Review by JacobCallahan — APPROVED** [review]

**Pull request merged**